dspace-api/src/main/java/org/dspace/content/dao/impl/MetadataValueDAOImpl.java

```diff
@@ -49,6 +49,17 @@ protected MetadataValueDAOImpl() {
         return list(context, criteriaQuery, false, MetadataValue.class, -1, -1);
     }
 
+    @Override
+    public Iterator<MetadataValue> findByFieldAndValue(Context context, MetadataField metadataField, String value)
+            throws SQLException {
+        String queryString = "SELECT m from MetadataValue m where metadata_field_id = :metadata_field_id " +
```

Comment:
Is the DB optimized for this type of queries?
Should this be limited to a specific DSO type (e.g. only items)?

Reply:
DB is not fully optimized for these, but limiting to DSO type (Items) should indeed improve performance. If this were performance-critical, I would advocate for possibly creating an index that involved the text_value, but that would be a rather high cost (space, time) so it would be important to show a large benefit. We will ensure the query *is* limited by DSO type. I also think we could stand to try this on a large test repository and see how ingest rate is affected by value-based vs rowName-based lookups.